src/main/java/melody/Melody.java

```diff
@@ -0,0 +1,157 @@
+package melody;
+
+import melody.command.CommandType;
+import melody.exception.MelodyException;
+import melody.parser.Parser;
+import melody.storage.Storage;
+import melody.task.*;
+import melody.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Main class
+ * Provides methods
+ */
+
+public class Melody {
+
+    private static TaskList tasks = new TaskList();
+    private static Storage storage = new Storage("./data/melody.Melody.txt");
+    private static Ui ui = new Ui();
+
+    public static void main(String[] args) {
+        ui.showWelcome();
+
+        try {
+            ArrayList<Task> loadedTasks = storage.loadTasks();
+            tasks = new TaskList(loadedTasks);
+            String wordy = tasks.size() == 1 ? " task" : " tasks" ;
+            ui.showMessage("Loaded " + tasks.size() + wordy + " from storage.");
+        } catch (Exception e) {
+            ui.showLoadingError(e.getMessage());
+            tasks = new TaskList();
+        }
+
+        Scanner scanner = new Scanner(System.in);
+        String input;
+
+        while (true) {
+
+            try {
+                input = ui.readCommand();
+                handleCommand(input);
+
+            } catch (MelodyException e) {
+                ui.showError(e.getMessage());
+            } catch (Exception e) {
+                ui.showError("  ☹ Oops! Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Generates a response for the user's chat message.
+     */
+    public String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    private static void handleCommand(String input) throws MelodyException {
+        CommandType commandType = Parser.parseCommand(input);
+
+        switch (commandType) {
```

Comment:
Case clauses are to be indented at the same level as switch statement. Maybe you could remove the indentation? 
```suggestion
        switch (commandType) {
        case BYE:
```